src/openassetio-python/tests/package/hostApi/test_entityreferencepager.py

```diff
@@ -137,6 +137,13 @@ def makePagerInScope():
         # destroyed.
         assert weak_interface_ref() is None
 
+    def test_when_EntityReferencePager_destructed_close_is_called(
```

Comment:
[fix] Could do with a test for when an exception is thrown in `close`.

I don't think it's possible to test the non-`exception` exception case from Python, but we could at least test the `std::exception` case.

Reply:
Yeah good point.

https://github.com/OpenAssetIO/OpenAssetIO/pull/1013/commits/5562006bb300f4ea094be3dda8dc339d86114e4f

Reply:
I tried doing a `raise 1` to see if I could throw an int but you're right, python complains that all exceptions must be exception types.